### PR TITLE
chore: rewrite transferOwnership script to use Foundry

### LINKED ISSUE #102: chore(legacy-core): forgify transfer authenticator proxy contract ownership script

### LINKED ISSUE #103: chore: remove hardhat tasks

### PR DESCRIPTION
**Pull request by fedgiac**
## Description

The transferOwnership script has been ported to Foundry, with feature parity, including tests.

There was the need for a few workarounds in the tests, mostly because `prank` and `broadcast` aren't compatible with each other and because our proxy is based on Solidity 0.7 and its code isn't directly available in this repo.

## Test Plan

New tests.
I also deployed a new set of contracts on Sepolia and used the script to change the owner and manager. You can see the two resulting transactions for the [manager](https://sepolia.etherscan.io/tx/0xe4d4499c66f696c671374ffc3f7255dc05fc09b9643af85e4c0e4243919f044a#eventlog) and the [owner](https://sepolia.etherscan.io/tx/0x42a592e04a3a76463b51bff67f38a7195875400e41b7b59737ac8dd41dcb45a7#eventlog).

## Related Issues

Closes #102
Closes #103

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,7 +2,7 @@ build/
 coverage/
 deployments/*/solcInputs
 deployments/localhost/
-lib/
+/lib/
```
[the remark this answers is not in the document]
Comment (fedgiac) [reply .gitignore:5]: I created a few subfolders named `lib/` in this PR to collect utility contracts that aren't test or scripts.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -57,7 +57,7 @@ jobs:
         run: |
           forge --version
           if [ "${{ matrix.profile }}" == "solc-0.7.6" ]; then
-            forge build --sizes --use 0.7.6 --skip 'test/*'
+            forge build --sizes --use 0.7.6 --skip 'test/*' --skip 'src/scripts/*'
```
[the remark this answers is not in the document]
Comment (fedgiac) [reply .github/workflows/test.yml]: Done. I added an extra build step on CI to make sure files always compile.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -63,6 +63,11 @@ jobs:
           fi
         id: build
 
+      - name: Run Forge build for scripts
+        if: matrix.profile != 'solc-0.7.6'
+        run: forge build --contracts ./scripts/
+        id: build-scripts
```
Comment (mfw78) [inline .github/workflows/test.yml:69]: I think this is a plurality issue. Forge by default includes `/script`, `/src`, and `/test` when building (as observed by `forge init temp && cd temp && forge build && ls -alh out/`).

**Review by mfw78 — COMMENTED** [review]
Seems plurality strike (sic) again. We should be putting scripts in `script` 🙈

**Review by mfw78 — APPROVED** [review]

**Pull request merged**